File: pe39.py
s=0
maxs=0
maxp=0
for p in range(1, 1001):
	for a in range(1, p):
		for b in range(a+1, p):
			c=p-a-b
			if(c>b and a+b+c==p and a+b>c and b+c>a and c+a>b and a*a+b*b==c*c):
				s+=1
			#print(p,a,b,c)
	if(s>maxs):
		maxs=s
		maxp=p

File: pe39.java
class pe34
{
	public static void main(String args[])
	{
		int sol,maxsol=0,maxp=0,a,b,c,p;
		for(p=1;p<=1000;p++)
		{
			sol=0;
			for(a=1;a<p;a++)
			{
				for(b=a+1;b<p;b++)
				{
					c=p-a-b;
					if(c>b && a+b>c && b+c>a && a+c>b && a*a+b*b==c*c)
						sol++;
				}
			}
			if(sol>maxsol)
			{
				maxsol=sol;
				maxp=p;
			}
		}
		System.out.print(maxp);
	}
}